packages/slycat/web/server/handlers.py

```diff
@@ -2759,6 +2758,128 @@ def post_smb_browse(hostname, path):
+
+    combined_dataset = []
+    separated_dataset = [] # This will get sent to create_project_data for saving the inputs and output separately
+    response_headers = []
+    variable_headers = []
+    attributes = []
+
+    column_headers_input = list(h5[input_path].dims[1][0])
+    column_headers_output = list(h5[output_path].dims[1][0])
+
+    # Once we have column headers, this is how we can get/store them. 
+    for i, column in enumerate(column_headers_input):
+        variable_headers.append(str(column.decode('utf-8')))
+        attributes.append({"name": str(column.decode('utf-8')), "type": str(type(unformatted_input[0][i])).split('numpy.')[1].split("'>")[0]})
+    for j, column in enumerate(column_headers_output):
+        response_headers.append(str(column.decode('utf-8')))
+        attributes.append({"name": str(column.decode('utf-8')), "type": str(type(unformatted_output[0][j])).split('numpy.')[1].split("'>")[0]})
+
+    combined_headers = numpy.concatenate((response_headers, variable_headers), axis=0)
+    combined_headers = combined_headers.tolist()
+    combined_data = numpy.concatenate((unformatted_input, unformatted_output), axis=1)
+    combined_data = numpy.transpose(combined_data)
+
+    combined_data = combined_data.tolist()
+
+    for row in combined_data:
+        combined_dataset.append(numpy.asarray(row))
+
+    dimensions = [{"name": "row", "type": "int64", "begin": 0, "end": len(combined_dataset[0])}]
+
+    array_index = 0
+    slycat.web.server.put_model_arrayset(database, model, 'data-table', input)
+    slycat.web.server.put_model_array(database, model, 'data-table', 0, attributes, dimensions)
+    slycat.web.server.put_model_arrayset_data(database, model, 'data-table', "%s/.../..." % array_index, combined_data)
+
+    return '...'
+
+def post_browse_hdf5(path, pid, mid):
+    def allkeys_single_level(obj, tree_structure):
+        path = obj.name # This is current top level path
+        tree_structure['path'] = path
+        tree_structure['name'] = []
+        tree_structure['sizes'] = []
+        tree_structure['types'] = []
+        tree_structure['mtimes'] = []
+        tree_structure['mime-types'] = []
+        all_items = obj.items()
+        for key, value in all_items:
+            # key will be all the sub groups and datasets in the current path
+            tree_structure['name'].append(key)
+            tree_structure['sizes'].append(0)
+            tree_structure['mtimes'].append('2024')
+            if isinstance(value, h5py.Group):
+                tree_structure['mime-types'].append('application/x-directory')
+                tree_structure['types'].append('d')
+            else:
+                tree_structure['mime-types'].append('file')
+                tree_structure['types'].append('f')
+        return tree_structure
+    # Need to find the HDF5 stored on Slycat server, so we can query it for the path.
+    path = path.replace('-', '/')
```

Comment:
I would map this to something like hdf5_path instead of writing over the passed in function arg